### ISSUE TITLE
Missing case for power set to None in plane_wave field function

### ISSUE THREAD

**Issue by AaronRiSa** [chromatix-team/chromatix#170]
When creating a gaussian plane wave or generic field one can set power to None and get the normalized field. That case is missing for the plane wave function (@diptodip, @GJBoth):

```python
probe_flat = cx.plane_wave(
    shape,
    dx,
    wavelength,
    wavelength_ratio,
    power=None,
    pupil=lambda field: cx.circular_pupil(field, diameter),
)
```

Missing part in plane_wave definition: 
```python
    def plane_wave(
    shape: tuple[int, int],
    dx: ScalarLike,
    spectrum: ScalarLike,
    spectral_density: ScalarLike,
    power: ScalarLike | None = 1.0,
    amplitude: ScalarLike = 1.0,
    kykx: ArrayLike | tuple[float, float] = (0.0, 0.0),
    pupil: FieldPupil | None = None,
    scalar: bool = True,
) -> ScalarField | VectorField:

    ...

    if pupil is not None:
        field = pupil(field)
    if power is not None:
        field = field * jnp.sqrt(power / field.power)
    return field

```
